### PR TITLE
Default settings, presets and configuration for dynamic shadows

### LINKED ISSUE #12260: State that dynamic shadows, even if activated, need games/mods to opt-in to be displayed
See https://github.com/minetest/minetest/issues/12259#issuecomment-1114668138, possibly underneath the shadows option in the main menu

### PR DESCRIPTION
**Pull request by x2048**
This PR tunes the way dynamic shadows are configured on the client:
* Fixes #12260 
* Tunes the default parameter values
* Tunes the quality presets

The new presets follow these ratios with every quality level:
* Texture resolution is doubled (x2)
* Max shadow distance is multiplied by 1.5

The result is that pixel density and shadow quality near player also increases with every preset.

## To do

This PR is a Work in Progress

## How to test

1. Start minetest
2. Go to settings, enable shaders
3. Change the presets for dynamic shadows.

### CODE REVIEW

**Comment by x2048** [comment]
Modified look:
image

**Comment by kilbith** [comment]
Give me [that](https://github.com/x2048/minetest/tree/shadows_frequency) instead and get rid of the shadow update frames bullshit.

**Comment by x2048** [comment]
I've made two polls on Discord, here are the results:

image

image

I have reduced maximum rendering distance for many presets based on this feedback, to favor more quality over coverage. I also doubled shadow map resolution for High and Ultra High, going for 8192 in Ultra High.

**Comment by lhofhansl** [comment]
I don't think we even need that max-shadow distance as config option.
We can always make it view_range * 1.5.

**Comment by x2048** [comment]
@lhofhansl Going to very high view range will kill the shadow quality near the player, and shadow distance is not a priority for high-quality preset. I still think we should have a limit for shadow range.

**Comment by lhofhansl** [comment]
So with #12364 and can leave shadow_update_frames the way it is, right?

File: builtin/settingtypes.txt [hunk 1]
```diff
@@ -626,8 +626,8 @@ shadow_map_color (Colored shadows) bool false
 #    Spread a complete update of shadow map over given amount of frames.
 #    Higher values might make shadows laggy, lower values
 #    will consume more resources.
-#    Minimum value: 1; maximum value: 16
-shadow_update_frames (Map shadows update frames) int 8 1 16
+#    Minimum value: 1; maximum value: 8
+shadow_update_frames (Map shadows update frames) int 1 1 8
```
Comment (lhofhansl) [inline builtin/settingtypes.txt:630]: 1 has quite a performance difference on my machine (with Intel or NVidia cards).
With #12364 we can leave this as is I think. (both default and max value)

Comment (x2048) [reply builtin/settingtypes.txt]: Done

**Comment by x2048** [comment]
So, this is the current proposal. The presets are as follows:

| Preset | Texure resolution | Maximum Range | Filtering |  Effects
| -------- | ----------------- | --------------------------- | ------------ | -----------
| Very Low | 512 | 62 | No filtering | None
| Low | 1024 | 93 | No filtering | None
| Medium | 2048 | 140 | 3x3 / 9 | None
| High | 4096 | 210 | 5x5 / 25 | Colored Shadows
| Very High | 8192 | 300 | 5x5 / 25 | Colored Shadows

Comments, feedback are welcome.

**Comment by lhofhansl** [comment]
Hard to say whether these pre-sets are good for everyone. These seem to be good enough.

**Comment by lhofhansl** [comment]
I still would like if we could derive more things: For example the max shadow distance and shadow map resolution from the view distance instance of specific config options... But I happily take this.

**Review by SmallJoker — APPROVED** [review]
Looks good; works.

**Pull request merged**